src/extractor/guidance/coordinate_extractor.cpp

```diff
@@ -132,7 +132,7 @@ util::Coordinate CoordinateExtractor::ExtractRepresentativeCoordinate(
     {
         const auto result = ExtractCoordinateAtLength(
             skipping_inaccuracies_distance, coordinates);
-        BOOST_ASSERT(is_valid_result(coordinates.back()));
+        BOOST_ASSERT(is_valid_result(result));
```

Comment:
This asserts the result is neither the to nor the intersection node? The assertion before therefore could trigger at any time?

Should this be backported to 5.5?